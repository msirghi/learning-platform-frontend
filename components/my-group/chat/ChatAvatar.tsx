import React from 'react';
import styles from '../../../styles/modules/MyGroup.module.scss';

export const ChatAvatar: React.FC = () => {
  return <div className={styles.avatar}></div>;
};
