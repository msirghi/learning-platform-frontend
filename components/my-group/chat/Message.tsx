import React from 'react';
import { ChatMessage } from '../../../common/types';
import styles from '../../../styles/modules/MyGroup.module.scss';
import dateformat from 'dateformat';
import { ChatAvatar } from './ChatAvatar';

type Props = {
  chatMessage: ChatMessage;
  fromMe?: boolean;
  hideAvatar?: boolean;
  fontSize: number;
};

export const Message: React.FC<Props> = ({
  hideAvatar,
  chatMessage: { author, date, message },
  fromMe,
  fontSize
}) => {
  return (
    <div className={`${styles.messageContainer} ${fromMe ? styles.myMessage : undefined}`}>
      <div
        className={`${styles.messageContentContainer} ${
          fromMe ? styles.myMessageContainer : undefined
        }`}
      >
        {!fromMe && !hideAvatar && <ChatAvatar />}
        {fromMe && <div className={styles.date}>{dateformat(date, 'h:MM')}</div>}
        <div
          className={`${styles.content} ${
            fromMe ? styles.myMessageContent : styles.notMyMessageContent
          }`}
        >
          <div style={{ fontSize }}>{message}</div>
          {/* <div className={styles.author}>{author}</div> */}
        </div>
        {fromMe && !hideAvatar && <ChatAvatar />}
        {!fromMe && <div className={styles.date}>{dateformat(date, 'h:MM')}</div>}
        {/* <div className={styles.author}>{author}</div> */}
      </div>
    </div>
  );
};
