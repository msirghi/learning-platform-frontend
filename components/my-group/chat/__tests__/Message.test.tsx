import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { ChatAvatar } from '../ChatAvatar';
import { Message } from '../Message';

describe('ChatMessage component', () => {
  const defaultProps = {
    hideAvatar: true,
    chatMessage: { date: new Date(), message: 'Message', author: 'Me', id: 1 },
    fromMe: false,
    fontSize: 14
  };

  it('should match the snapshot if all the props are provided', () => {
    const wrapper = shallow(<Message {...defaultProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render ChatAvatar component', () => {
    const wrapper = shallow(<Message {...defaultProps} fromMe={false} hideAvatar={false} />);
    expect(wrapper.find(ChatAvatar)).toHaveLength(1);
  });

  it('should render ChatAvatar component if the message is mine and avatar is hiddedn', () => {
    const wrapper = shallow(<Message {...defaultProps} fromMe hideAvatar={false} />);
    expect(wrapper.find(ChatAvatar)).toHaveLength(1);
  });
});
