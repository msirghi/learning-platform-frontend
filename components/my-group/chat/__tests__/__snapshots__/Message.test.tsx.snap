// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage component should match the snapshot if all the props are provided 1`] = `
<div
  className="messageContainer undefined"
>
  <div
    className="messageContentContainer undefined"
  >
    <div
      className="content notMyMessageContent"
    >
      <div
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        Message
      </div>
    </div>
    <div
      className="date"
    >
      2:00
    </div>
  </div>
</div>
`;
